### ISSUE TITLE
livedata server exception

### ISSUE THREAD

**Issue by simonbelanger** [cult-of-coders/redis-oplog#206]
I see a lot of these exceptions in the server log, any chance it might be related to `redis-oplog` : 

![2017-11-10 at 11 37](https://user-images.githubusercontent.com/17987029/32668523-e126e28c-c60b-11e7-92ac-57d7b28f05df.png)

**Comment by theodorDiaconu** [comment]
@simonbelanger it seems that yes:

This runs inside Session.close()
```
     self._deactivateAllSubscriptions(); // Defer calling the close callbacks, so that the caller closing
```

```
_deactivateAllSubscriptions: function () {
    var self = this;

    _.each(self._namedSubs, function (sub, id) {
      sub._deactivate();
    });

    self._namedSubs = {};

    _.each(self._universalSubs, function (sub) {
      sub._deactivate();
    });

    self._universalSubs = [];
  },
```

Inside: _.extend(Subscription.prototype, {})
```
// This calls all stop callbacks and prevents the handler from updating any
  // SessionCollectionViews further. It's used when the user unsubscribes or
  // disconnects, as well as during setUserId re-runs. It does *NOT* send
  // removed messages for the published objects; if that is necessary, call
  // _removeAllDocuments first.
  _deactivate: function () {
    var self = this;
    if (self._deactivated) return;
    self._deactivated = true;

    self._callStopCallbacks();

    Package.facts && Package.facts.Facts.incrementServerFact("livedata", "subscriptions", -1);
  },
```

But I don't get it, in Redis we don't create a custom subscription, we either return nonEligibleCursors, or nothing, so the subscription is properly handled by Redis.

Have you played with` publishWithRedis:false` by any chance ? Last time there weren't these issues if I remember.

**Comment by simonbelanger** [comment]
> Have you played with publishWithRedis:false by any chance ?

no

**Comment by theodorDiaconu** [comment]
Try pulling from master. See if it still happens. It may be related to disable mergebox package used for vent. But i removed bc i found a better way.

**Comment by theodorDiaconu** [comment]
Did you solve this @simonbelanger  ?

**Comment by simonbelanger** [comment]
I didn't have the time to try the fix mentionned... we're gonna put this on ice for the next days, until we have time to investigate what happened on the servers.

**Comment by Maxpain177** [comment]
Same error on latest commit!

**Comment by theodorDiaconu** [comment]
@Maxpain177 can you help me reproduce this ?

**Comment by Maxpain177** [comment]
@theodorDiaconu Yes. But later, i need to catch this error on production server in --inspect mode.

**Comment by Maxpain177** [comment]
@theodorDiaconu, Sorry, i fixed this error. I just add names to all publications without name (Meteor.publish(null)). It's also caused memory leaks.

**Comment by theodorDiaconu** [comment]
@Maxpain177 so you're saying we get this problem because of null publications ? Also. What is causing the memory leak ?

**Comment by Maxpain177** [comment]
@theodorDiaconu Yes exactly, adding names to unnamed publications fixed an error and memory leak.
Unfortunately i don't know how to catch memory leaks :(

**Comment by theodorDiaconu** [comment]
very very strange, can't explain this currently, but I think this ticket should remain active to do more extensive research on global publications.

**Comment by Maxpain177** [comment]
@theodorDiaconu I also noticed that global publications, unlike publications has names, called on server startup. Maybe this will help you.

**Comment by Maxpain177** [comment]
Hmmm.. Seems like problem is still exists, but errors appears quite rarely.

**Comment by theodorDiaconu** [comment]
@Maxpain177 is there any way you can reliably reproduce this ?

**Comment by simonbelanger** [comment]
@Maxpain177 I'd also be interested to know how to reproduce this (if you were able to). How did you manage to link the error to unnamed publications?
